cmd/zgenhostid/zgenhostid.c

```diff
@@ -88,20 +115,23 @@ main(int argc, char **argv)
 			exit(EXIT_FAILURE);
 		}
 
-		if (input_i < 0x1 || input_i > 0xFFFFFFFF) {
+		if (input_i < 0x1 || input_i > UINT32_MAX) {
 			fprintf(stderr, "%s\n", strerror(ERANGE));
 			usage();
 		}
 	}
 
 	struct stat fstat;
-	if (forcew == 0 && stat(path, &fstat) == 0 &&
+	if (force_fwrite == 0 && stat(path, &fstat) == 0 &&
 	    S_ISREG(fstat.st_mode) && fstat.st_size == 4) {
```

Comment:
I can imagine sysadmins not noticing the size of /etc/hostid files and being confused that sometimes zgenhostid will replace them, and other times will not.

Reply:
so what's the preferred way? just check for presence and `S_ISREG`, removing size check?